tests/test_validators.py

```diff
@@ -111,30 +173,30 @@ class Model(BaseModel):
         b: int
         a: List[int]
 
+        # TODO: do we support `each_item`?
```

Comment:
Does this currently work? Just curious

Reply:
It doesn't work. `@validator` didn't accept an `each_item` param before this PR and this PR is not adding it. As discussed in https://github.com/pydantic/pydantic/issues/4707 I think we should actually not support this and instead make you apply validators to child items using `Annotated`.

Reply:
Oops I missed the skipped lines and thought this test had its `xfail` removed.

I added a note about this to the migration guide points ("Describe how to replicate the functionality of each_item validators").

Does applying the validators to the child items using `Annotated` work now? Just thinking we could update this test to do that, or at least update the `TODO:` comment to say that the test should eventually be updated to do something like that.

Reply:
Not yet. That needs us to create the `PlainValidator` and others metadata.

One important thing to note is that the item validators won't have access to `values: Dict[str, Any]` or the model class. I think that seems like a reasonable limitation. We have the validation context where if you want to do something weird like that you can.

Reply:
Yeah I think it's reasonable for us to say that if you want to access that stuff, you just have to define the function in a way that it applies to the "full" field value, rather than individual items within it.